defaults/php_config_ref.yml

```diff
@@ -0,0 +1,48 @@
+---
+# define parameters depending for the php version installed
+# for any new version, only specify keys with values different from defaults
+# defaults should work for php > 7.3
+defaults:
+  php_dir: "/etc/php/{{ php_ver }}"
+  php_bin: "php-fpm{{ php_ver }}"
+  php_pkg_apcu: "php{{ php_ver }}-apcu"
+  php_pkg_spe:
+    - "php{{ php_ver }}-bcmath"
+    - "php{{ php_ver }}-gmp"
+    - "php{{ php_ver }}-imagick"
+    - "php{{ php_ver }}-mbstring"
+    - "php{{ php_ver }}-redis"
+    - "php{{ php_ver }}-xml"
+    - "php{{ php_ver }}-zip"
+  php_socket: "/run/php/php{{ php_ver }}-fpm.sock"
+
+7_3:
```

Comment:
I would even remove everything from this line on and replace with 2 tasks:

if php_ver <= 7_3 you need to install:
    - "php-imap"
    - "php-imagick"
    - "php-redis"
    - "php{{ php_ver }}-xml"
    - "php{{ php_ver }}-zip"
    - "php{{ php_ver }}-mbstring"

If it is 7.0 you add a new task just to add "php7.0-mcrypt" -> this task should be removed soon as Nextcloud is phasing out support vor versions with php 7.0.

Reply:
Thank you for feedback @staticdev 👍 
I understand your point: we could go even further in automation by templating even more the reference configuration for PHP.
I won't add tasks for managing special cases. The goal of the pr is to remove them and reduce   the tasks to only "work" things.
I think templating the supported php conf  will make the role harder to apprehend for newcomers.
Simply stating ( not that much ) config keys that change from default will look simpler than a bunch of if blocks in the defaults key.
It's a compromise between  ease of maintenance  and features I think.

I'll make a few more change to remove more tasks in setup_env in this PR
Do not hesitate to counter me on that.
Regards.
Aal

Reply:
@aalaesar this is ok, then I would say we try to merge this PR before introducing new changes. It already has a good cleanup size. In this case I would recommend adding new changes in a new PR if it is not too annoying for you.